library/core/src/ptr/mut_ptr.rs

```diff
@@ -35,7 +35,10 @@ impl<T: ?Sized> *mut T {
     pub const fn is_null(self) -> bool {
         // Compare via a cast to a thin pointer, so fat pointers are only
         // considering their "data" part for null-ness.
-        (self as *mut u8).guaranteed_eq(null_mut())
+        match (self as *mut u8).guaranteed_eq(null_mut()) {
+            None => false,
+            Some(res) => res,
+        }
```

Comment:
```suggestion
        (self as *mut u8).guaranteed_eq(null_mut()).unwrap_or(false)
```

Reply:
that doesn't work, this is a const fn :)

Reply:
Argh...